hack/local-up-karmada-helm.sh

```diff
@@ -0,0 +1,183 @@
+# the key of the map is the cluster name / context name of your member clusters
+# the value of the map is the corresponding kubeconfig path of the member cluster
+# if you already has member clusters, replace the key with real name and replace the value with real kubeconfig path
+# you can add any number pull mode clusters as you expect, just append this map as following examples
+declare -A PULL_MODE_MEMBERS
+PULL_MODE_MEMBERS["member2"]="${HOME}/.kube/members.config"
+
+echo "########## start installing karmada control plane ##########"
+
+# 1. create KinD cluster if you set NEED_CREATE_KIND_CLUSTER
+if ${NEED_CREATE_KIND_CLUSTER}; then
+  # 1.1 create karmada-host cluster
+  kind delete clusters karmada-host
+  rm -rf ~/.karmada "${KARMADA_HOST_KUBECONFIG}"
+  hack/create-cluster.sh ${KARMADA_HOST_NAME} "${KARMADA_HOST_KUBECONFIG}"
+fi
+
+# 2. fetch images in advance is you set IMAGE_FROM
+if [ "${IMAGE_FROM}" == "pull-in-advance" ]; then
+  ## 2.1 use 'docker pull' to fetch target images to local node in advance
+  imgs=$(cat charts/karmada/values.yaml | grep -C 1 'repository:' | sed 's/*karmadaImageVersion/latest/g' | awk -F ':' '{print $2}' | sed 's/\"//g' | xargs -n3 | awk '{print $1"/"$2":"$3}')
+  for img in ${imgs}; do
+    docker pull "${img}"
+  done
+elif [ "${IMAGE_FROM}" == "make" ]; then
+  ## 2.2 build karmada images from source code with latest tag and pull images (other than karmada) in advance
+  export VERSION="latest"
+  export REGISTRY="docker.io/karmada"
+  make images GOOS="linux" .
+
+  imgs=$(cat charts/karmada/values.yaml | grep -v 'karmada' | grep -C 1 'repository: ' | sed 's/*karmadaImageVersion/latest/g' | awk -F ':' '{print $2}' | sed 's/\"//g' | xargs -n3 | awk '{print $1"/"$2":"$3}')
+  for img in ${imgs}; do
+    docker pull "${img}"
+  done
+fi
+
+# 3. load images into KinD karmada-host cluster in advance if you set LOAD_IMAGE_IN_ADVANCE
+if ${LOAD_IMAGE_IN_ADVANCE}; then
+  imgs=$(cat charts/karmada/values.yaml | grep -C 1 'repository:' | sed 's/*karmadaImageVersion/latest/g' | awk -F ':' '{print $2}' | sed 's/\"//g' | xargs -n3 | awk '{print $1"/"$2":"$3}')
+  for img in ${imgs}; do
+    kind load docker-image "${img}" --name ${KARMADA_HOST_NAME}
+  done
+fi
+
+# 4. this script try to deploy karmada-apiserver by host-network
+# so, it needs to get host-network ip (node ip) from kube-apiserver, and then add this ip to values.yaml as SANs of certificate
+export KUBECONFIG=${KARMADA_HOST_KUBECONFIG}
+HOST_IP=$(kubectl get ep kubernetes -o jsonpath='{.subsets[0].addresses[0].ip}')
+sed -i'' -e "/localhost/p; s/localhost/${HOST_IP}/" charts/karmada/values.yaml
+
+# 5. install karmada at karmada-host cluster by helm
+helm install karmada -n karmada-system \
+        --kubeconfig "${KARMADA_HOST_KUBECONFIG}" \
+        --create-namespace \
+        --dependency-update \
+        --set apiServer.hostNetwork=true,components={"search,descheduler"} \
```

Comment:
How about making the component a configurable optional parameter and empty by default.

Reply:
nice